lib/pure/osproc.nim

```diff
@@ -410,6 +410,45 @@ when defined(Windows) and not defined(useNimRtl):
       copyMem(addr(result[L]), cstring(x), x.len+1) # copy \0
       inc(L, x.len+1)
 
+  proc envToWideCString(env: StringTableRef): WideCString =
+    # newWideCString stops on \0 characters, so we have to
+    # convert every pair separately.
+    const wcharSize = 2
+    var rows = newSeq[tuple[str: WideCString, len: int]]()
```

Comment:
Can't see a way to do it without it, that doesn't involve calling newWideCString twice per pair, or allocating overly large result string.